k4MarlinWrapper/src/components/EDM4hep2Lcio.cpp

```diff
@@ -212,6 +212,29 @@ void EDM4hep2LcioTool::convertReconstructedParticles(
   lcio_event->addCollection(conv_recops, lcio_coll_name);
 }
 
+// Transfer info from EDM4hep EventHeader to LCIO event
+void EDM4hep2LcioTool::convertEventHeader(const std::string& e4h_coll_name, lcio::LCEventImpl* lcio_event) {
+  DataHandle<edm4hep::EventHeaderCollection> header_handle{e4h_coll_name, Gaudi::DataHandle::Reader, this};
+  const auto                                 header_coll = header_handle.get();
+
+  const auto& event_n      = header_coll->eventNumber();
+  const auto& run_n        = header_coll->runNumber();
+  const auto& timestamp    = header_coll->timeStamp();
+  const auto& event_weight = header_coll->weight();
```

Comment:
You are calling the "vectorized access functionality here". It would be much easier to simply to make sure the collection has only one element and then retrieve the first element from the collection and use the `get` prefixed functions to handle only one event header, sth like:
```cpp
lcio_event->setEventNumber(evtHeader.getEventNumber());
// ...
```

Reply:
thanks, I was not aware of that possibility, much easier indeed :)